planning/behavior_path_planner/src/behavior_path_planner_node.cpp

```diff
@@ -607,6 +609,35 @@ void BehaviorPathPlannerNode::run()
     hazard_signal.stamp = get_clock()->now();
     turn_signal_publisher_->publish(turn_signal);
     hazard_signal_publisher_->publish(hazard_signal);
+
+    const auto intersection_result = turn_signal_decider_.getIntersectionTurnSignalFlag();
+    const bool intersection_flag = intersection_result.first;
+    const bool approaching_intersection_flag = intersection_result.second;
+    if (intersection_flag || approaching_intersection_flag) {
+      const uint16_t steering_factor_direction = std::invoke([&turn_signal]() {
+        if (turn_signal.command == TurnIndicatorsCommand::ENABLE_LEFT) {
+          return SteeringFactor::LEFT;
+        }
+        return SteeringFactor::RIGHT;
+      });
+
+      const auto intersection_pose_distance = turn_signal_decider_.getIntersectionPoseAndDistance();
+
+      if (intersection_flag) {
+        steering_factor_interface_ptr_->updateSteeringFactor(
+          {intersection_pose_distance.first, intersection_pose_distance.first},
+          {intersection_pose_distance.second, intersection_pose_distance.second},
+          SteeringFactor::INTERSECTION, steering_factor_direction, SteeringFactor::TURNING, "");
+      } else {
+        steering_factor_interface_ptr_->updateSteeringFactor(
+          {intersection_pose_distance.first, intersection_pose_distance.first},
+          {intersection_pose_distance.second, intersection_pose_distance.second},
+          SteeringFactor::INTERSECTION, steering_factor_direction, SteeringFactor::TRYING, "");
+      }
+    } else {
+      steering_factor_interface_ptr_->clearSteeringFactors();
+    }
+    steering_factor_interface_ptr_->publishSteeringFactor(get_clock()->now());
```

Comment:
@tkhmy Could you functionalize L613-L640 for readability?

Reply:
changed in [1aaa948](https://github.com/autowarefoundation/autoware.universe/pull/1706/commits/1aaa9484d64b93e4d195e312f3f1a6327acbfa56)